src/main/java/games/strategy/engine/data/AllianceTracker.java

```diff
@@ -65,12 +65,10 @@ protected void addToAlliance(final PlayerID player, final String allianceName) {
    * @return a set of all the games alliances, this will return an empty set if you aren't using alliances.
    */
   public Set<String> getAlliances() {
-    final Iterator<PlayerID> keys = this.alliances.keySet().iterator();
-    final Set<String> alliances = new HashSet<>();
-    while (keys.hasNext()) {
-      alliances.addAll(this.alliances.get(keys.next()));
-    }
-    return alliances;
+    return alliances.keySet().stream()
+        .map(alliances::get)
+        .flatMap(Collection::stream)
+        .collect(Collectors.toSet());
```

Comment:
Or  even
```java
return new HashSet<>(alliances.values());
```
But for most of the code I didn't want to change any functionality in order to avoid any regressions